tests/python/multidevice/test_deepseek_v3.py

```diff
@@ -62,12 +69,77 @@ def load_model(config: transformers.PretrainedConfig) -> transformers.PreTrained
     return transformers.AutoModel.from_config(config, trust_remote_code=True)
 
 
+class Executor(Enum):
+    # https://docs.pytorch.org/docs/stable/distributed.tensor.parallel.html
+    TORCH_TP = auto()
+    NVFUSER = auto()
+
+
+def parallelize_linear_with_nvfuser(
+    linear: torch.nn.Linear,
+    mesh: dist.device_mesh.DeviceMesh,
+    parallel_style: ParallelStyle,
+) -> torch.nn.Linear:
+    assert isinstance(linear, torch.nn.Linear), f"Unsupported layer: {linear}"
+
+    assert len(parallel_style.input_layouts) == 1, "Expect 1D mesh"
+    input_layout = parallel_style.input_layouts[0]
+
+    assert len(parallel_style.output_layouts) == 1, "Expect 1D mesh"
+    output_layout = parallel_style.output_layouts[0]
+
+    if isinstance(parallel_style, RowwiseParallel):
+        assert input_layout.is_shard(-1), "We only implemented TP."
+        assert output_layout.is_replicate(), "We only implemented TP."
+        return TensorParallelLinear.distribute(
+            linear, mesh, [input_layout], [Shard(-1)]
+        )
+
+    if isinstance(parallel_style, ColwiseParallel):
+        assert input_layout.is_replicate(), "We only implemented TP."
+        assert output_layout.is_shard(-1), "We only implemented TP."
+        return TensorParallelLinear.distribute(linear, mesh, [input_layout], [Shard(0)])
+
+    assert False, f"Unsupported parallel style: {parallel_style}"
+
+
+def parallelize_module_with_nvfuser(
+    module: torch.nn.Module,
+    mesh: dist.device_mesh.DeviceMesh,
+    parallel_plan: dict[str, ParallelStyle],
+    fqn: str,  # stands for fully qualified name
+    parent_module: Optional[torch.nn.Module] = None,
+):
```

Comment:
Can we code comment what this function is doing: Recursively find all the linear modules, and parallelize them with nvfuser if a parallel_plan is found